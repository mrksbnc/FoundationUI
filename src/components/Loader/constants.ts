export enum SpinnerVariant {
	Default = 'default',
	Info = 'info',
	Tertiary = 'tertiary',
	Success = 'success',
	Warning = 'warning',
	Danger = 'danger',
}

export enum SpinnerSize {
	XXS = 'xxs',
	XS = 'xs',
	SM = 'sm',
	MD = 'md',
	LG = 'lg',
	XL = 'xl',
	XXL = 'xxl',
	XXXL = 'xxxl',
}
