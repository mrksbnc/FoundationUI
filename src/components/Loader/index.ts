export { default as DsLoadingSpinner } from './DsLoadingSpinner.vue';
export { LoadingSpinnerColor, LoadingSpinnerSize } from './constants';
