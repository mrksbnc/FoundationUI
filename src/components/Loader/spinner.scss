@import '@/styles/layout.scss';

.bo-loading-spinner {
	position: absolute;
	border-radius: 50%;
	border: 2px solid transparent;
	animation: spin 2s linear infinite;
	-webkit-animation: spin 2s linear infinite;

	&.xxs {
		width: 8px;
		height: 8px;
	}

	&.xs {
		width: 12px;
		height: 12px;
	}

	&.sm {
		width: 14px;
		height: 14px;
	}

	&.md {
		width: 16px;
		height: 16px;
	}

	&.lg {
		width: 24px;
		height: 24px;
	}

	&.xl {
		width: 32px;
		height: 32px;
	}

	&.xxl {
		width: 40px;
		height: 40px;
	}

	&.xxxl {
		width: 48px;
		height: 48px;
	}

	&.default {
		border-top: 2px solid var(--black);
	}

	&.info {
		border-top: 2px solid var(--blue-600);
	}

	&.tertiary {
		border-top: 2px solid var(--gray-600);
	}

	&.success {
		border-top: 2px solid var(--green-600);
	}

	&.warning {
		border-top: 2px solid var(--yellow-500);
	}

	&.danger {
		border-top: 2px solid var(--red-600);
	}
}

@-webkit-keyframes spin {
	0% {
		-webkit-transform: rotate(0deg);
	}
	100% {
		-webkit-transform: rotate(360deg);
	}
}

@keyframes spin {
	0% {
		transform: rotate(0deg);
	}
	100% {
		transform: rotate(360deg);
	}
}
