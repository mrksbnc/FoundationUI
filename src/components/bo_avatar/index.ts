export * from './bo_avatar.contant';
export * from './bo_avatar.type';
export { default as BoAvatar } from './BoAvatar.vue';
