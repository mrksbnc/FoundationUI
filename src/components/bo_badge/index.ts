export { default as BoBadge } from './BoBadge.vue'
export * from './constants'
export * from './types'
