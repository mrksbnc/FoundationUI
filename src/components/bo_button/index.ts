export { default as BoButton } from './BoButton.vue';
export {
	BoButtonBorderRadiusClasses,
	BoButtonDarkFilledColorClasses,
	BoButtonDarkLinkTypeClasses,
	BoButtonDarkOutlineColorClasses,
	BoButtonHeightClasses,
	BoButtonLightFilledColorClasses,
	BoButtonLightLinkTypeClasses,
	BoButtonLightOutlineColorClasses,
	BoButtonLightShadowClasses,
	BoButtonPaddingClasses,
	BoButtonTextSizeClasses,
	BoButtonType,
} from './bo_button';
export type { BoButtonComponentProps } from './bo_button.types';
