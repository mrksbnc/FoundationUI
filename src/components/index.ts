export * from '@/components/bo_badge';
export * from '@/components/bo_button';
export * from '@/components/bo_icon';
export * from '@/components/bo_spinner';
