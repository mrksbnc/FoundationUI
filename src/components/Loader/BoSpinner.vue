<template>
	<span :class="['bo-loading-spinner', css]"></span>
</template>

<script setup lang="ts">
import { computed, toRefs, type PropType } from 'vue';
import { SpinnerVariant, SpinnerSize } from '@/components/Loader';
import type { OptionalCss } from '@/types';

const props = defineProps({
	variant: {
		type: String as PropType<SpinnerVariant>,
		default: () => SpinnerVariant.Default,
	},
	size: {
		type: String as PropType<SpinnerSize>,
		default: () => SpinnerSize.MD,
	},
});

const { size, variant } = toRefs(props);

const css = computed<OptionalCss>(() => {
	const classes: OptionalCss = {};

	switch (variant.value) {
		case SpinnerVariant.Default:
			classes['default'] = true;
			break;
		case SpinnerVariant.Info:
			classes['info'] = true;
			break;
		case SpinnerVariant.Success:
			classes['success'] = true;
			break;
		case SpinnerVariant.Warning:
			classes['warning'] = true;
			break;
		case SpinnerVariant.Danger:
			classes['danger'] = true;
			break;
		case SpinnerVariant.Tertiary:
			classes['tertiary'] = true;
			break;
	}

	switch (size.value) {
		case SpinnerSize.XS:
			classes['xs'] = true;
			break;
		case SpinnerSize.SM:
			classes['sm'] = true;
			break;
		case SpinnerSize.MD:
			classes['md'] = true;
			break;
		case SpinnerSize.LG:
			classes['lg'] = true;
			break;
		case SpinnerSize.XL:
			classes['xl'] = true;
			break;
	}

	return classes;
});
</script>

<style scoped lang="scss">
@import './spinner.scss';
</style>
