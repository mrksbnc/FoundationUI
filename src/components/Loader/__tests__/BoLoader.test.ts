import { BoSpinner, SpinnerSize, SpinnerVariant } from '@/components/Loader';
import { mount } from '@vue/test-utils';
import { describe, expect, it } from 'vitest';

describe('BoIcon', () => {
	it('renders properly', () => {
		const wrapper = mount(BoSpinner, {
			props: {
				size: SpinnerSize.MD,
				variant: SpinnerVariant.Info,
			},
		});
		expect(wrapper.html()).toMatchSnapshot();
	});
});
