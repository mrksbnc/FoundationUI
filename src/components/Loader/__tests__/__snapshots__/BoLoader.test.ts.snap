// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BoIcon > renders properly 1`] = `"<span data-v-0dc9c9ce=\\"\\" class=\\"bo-loading-spinner info md\\"></span>"`;
